src/track/track.h

```diff
@@ -363,6 +370,22 @@ class Track : public QObject {
             const mixxx::audio::StreamInfo& streamInfo);
 
   signals:
+    void artistChanged(const QString&);
+    void titleChanged(const QString&);
+    void albumChanged(const QString&);
+    void albumArtistChanged(const QString&);
+    void genreChanged(const QString&);
+    void composerChanged(const QString&);
+    void groupingChanged(const QString&);
+    void yearChanged(const QString&);
+    void trackNumberChanged(const QString&);
+    void trackTotalChanged(const QString&);
+    void commentChanged(const QString&);
+    void bpmTextChanged();
```

Comment:
Why do some signals carry the value and others not?

Reply:
Oh, even the docs don't apply a consistent approach:

https://doc.qt.io/qt-5/properties.html

Reply:
0 or 1 parameter are allowed, but if the new value is provided as a parameter than it's type must match the type of the property.

In order to keep the number of signals low let's apply the following rationale:
- For native properties without any derived properties provide the new value if possible
- For properties with derived properties use the name of the native property, reuse the signal for all derived properties, and don't provide any value.

By applying this rule we need to revert `bpmTextChanged` to `bpmChanged`.

Reply:
The `Track` object with its thread synchronization is a bastard and not really suitable for this purpose. What we actually would need is a distinction between *view model* and *domain model* classes.

Reply:
https://github.com/mixxxdj/mixxx/pull/3924/commits/640a59d095e959302b1b52428b7891f864c09961